src/cpu/aarch64/cpu_isa_traits.hpp

```diff
@@ -0,0 +1,158 @@
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_AARCH64_CPU_ISA_TRAITS_HPP
+#define CPU_AARCH64_CPU_ISA_TRAITS_HPP
+
+#include <type_traits>
+
+#include "dnnl_types.h"
+
+#include "common/dnnl_thread.hpp"
+#include "common/utils.hpp"
+
+/* in order to make selinux happy memory that would be marked with X-bit should
+ * be obtained with mmap */
+#define XBYAK_USE_MMAP_ALLOCATOR
+
+#include "cpu/aarch64/xbyak_aarch64/xbyak_aarch64.h"
+#include "cpu/aarch64/xbyak_aarch64/xbyak_aarch64_util.h"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+enum cpu_isa_bit_t : unsigned {
+    simdfp_bit = 1u << 0,
+    sve_128_bit = 1u << 1,
+    sve_256_bit = 1u << 2,
+    sve_384_bit = 1u << 3,
+    sve_512_bit = 1u << 4,
+};
+
+enum cpu_isa_t : unsigned {
+    isa_any = 0u,
+    simdfp = simdfp_bit,
+    sve_128 = sve_128_bit | simdfp,
+    sve_256 = sve_256_bit | simdfp,
+    sve_384 = sve_384_bit | simdfp,
+    sve_512 = sve_512_bit | simdfp,
+    isa_all = ~0u,
+};
+
+const char *get_isa_info();
+
+cpu_isa_t DNNL_API get_max_cpu_isa_mask(bool soft = false);
+status_t set_max_cpu_isa(dnnl_cpu_isa_t isa);
+dnnl_cpu_isa_t get_effective_cpu_isa();
+
+template <cpu_isa_t>
+struct cpu_isa_traits {}; /* ::vlen -> 32 (for avx2) */
+
+// pack struct so it can fit into a single 64-byte cache line
+#pragma pack(push, 1)
+struct palette_config_t {
+    uint8_t palette_id;
+    uint8_t startRow;
+    uint8_t reserved[14];
+    uint16_t cols[16];
+    uint8_t rows[16];
+};
+#pragma pack(pop)
+
+template <>
+struct cpu_isa_traits<isa_all> {
+    static constexpr dnnl_cpu_isa_t user_option_val = dnnl_cpu_isa_all;
+    static constexpr const char *user_option_env = "ALL";
+};
+
+template <>
+struct cpu_isa_traits<simdfp> {
+    typedef Xbyak_aarch64::VReg4S Vmm;
+    static constexpr int vlen_shift = 4;
+    static constexpr int vlen = 16;
+    static constexpr int n_vregs = 32;
+    static constexpr dnnl_cpu_isa_t user_option_val = dnnl_cpu_isa_simdfp;
+    static constexpr const char *user_option_env = "SIMD&FP";
+};
+
+template <>
+struct cpu_isa_traits<sve_512> {
+    typedef Xbyak_aarch64::ZRegS Vmm;
+    static constexpr int vlen_shift = 6;
+    static constexpr int vlen = 64;
+    static constexpr int n_vregs = 32;
+    static constexpr dnnl_cpu_isa_t user_option_val = dnnl_cpu_isa_sve_512;
+    static constexpr const char *user_option_env = "SVE_512";
+};
+
+inline const Xbyak_aarch64::util::Cpu &cpu() {
+    const static Xbyak_aarch64::util::Cpu cpu_;
+    return cpu_;
+}
+
+namespace {
+
+static inline bool mayiuse(const cpu_isa_t cpu_isa, bool soft = false) {
+    using namespace Xbyak_aarch64::util;
+
+    unsigned cpu_isa_mask = aarch64::get_max_cpu_isa_mask(soft);
+    if ((cpu_isa_mask & cpu_isa) != cpu_isa) return false;
+
+    switch (cpu_isa) {
+        case simdfp: return cpu().has(Cpu::tADVSIMD) && cpu().has(Cpu::tFP);
```

Comment:
This should be the case for all AArch64 targets.

Reply:
I preserved the option for `simdfp` -> `asimd`, because `ADVANCED_SIMD` is explicitly specified by MAX_CPU_ISA environment variable. https://github.com/oneapi-src/oneDNN/pull/850/commits/8b39d4d25bd9e999c343ea4086136f936926550d

Reply:
OK, thanks.